### ISSUE TITLE
Sulis stills are not displayed during replay

### ISSUE THREAD

**Issue by smccue2019** [WHOIGit/ndsf-sealog-client#27]
As part of the postdive process both a  .ppfx file and a collection of still images from Jason's highest res camera are added into the dive timeline, as events I believe.
The .ppfx file includes for each still a still name plus time plus naviation info (lat, lon, depth, HPR, alt).

During replay the events are placed within the timeline. However, the corresponding still is not displayed. Instead, an image blank is shown. This he actual image should be diusplayed with the event.

**Comment by rgov** [comment]
@smccue2019 Could you please point me to the script / process for loading the ppfx file?

**Comment by smccue2019** [comment]
On the shoreside server it's ~sealog/sealog-import-tools/import_lowering_into_sealog.sh.

It's something different at sea, and because my memory is fuzzy I've texted James for confirmation.
I believe that it's 'sealog-sulisCam-stills-import.py'. 'sealog_postdive.sh' also invokes some sulis-related actions, but IIRC it's used (after applying sulisCam-stills-import) to extract events including sulis events & stills for transport to shore.

**Comment by smccue2019** [comment]
I guess the above answer is a little light on detail. For full process refer to ~sealog/cruise_install_notes.txt.

TL/DR: modify cruise; modify lowering; import cruise; import lowering. The latter is what does the ingestion into the shoreside back end.

**Comment by smccue2019** [comment]
Note: the import script may have recently changed, because its behavior is not as I remember. I'm
exercising the behavior now as I install sealog for Jason cruise KM2212.

After invoking and choosing the cruise, the script asks the user to select the dive to import, or 'All'. The
latter is the typical choice. When selected, the script loops through each dive. However, for each dive the
user is asked to identify which vehicle produced the data.

The order of question at one time had vehicle selected earlier, and once identified it was not asked again. The
script then would loop through dives without stopping. The current script requires a user's attention throughout the
import, un-necessarily I would argue. A long cruise can occupy the better part of an hour of a user's time.

**Comment by rgov** [comment]
Makes sense, but can you capture that in a separate issue? Thanks.

On Nov 11, 2022, at 17:07, smccue2019 ***@***.***> wrote:

﻿

Note: the import script may have recently changed, because its behavior is not as I remember. I'm
exercising the behavior now as I install sealog for Jason cruise KM2212.

After invoking and choosing the cruise, the script asks the user to select the dive to import, or 'All'. The
latter is the typical choice. When selected, the script loops through each dive. However, for each dive the
user is asked to identify which vehicle produced the data.

The order of question at one time had vehicle selected earlier, and once identified it was not asked again. The
script then would loop through dives without stopping. The current script requires a user's attention throughout the
import, un-necessarily I would argue. A long cruise can occupy the better part of an hour of a user's time.

**Comment by mcschwartzman** [comment]
Keeping this thread up to date. I haven't yet attempted to dig through the import script, but looking at the db, it looks like what we're doing is creating Sulis Still events and directly linking the image filepath to those events. 

`{
		"event_value": "SulisCam",
		"event_options": [{
			"event_option_value": "/J2-1229/SulisCam/SulisCam.20191001_123550000.jpg",
			"event_option_name": "filename"
		}],
		"ts": "2019-10-01T12:35:50.000Z",
		"event_free_text": "",
		"event_author": "jason",
		"id": "5d93cafafef616043cc13d03"
	},`

This appears to be in contrast to other image data, which is actually stored in the event_aux_data collection. 
`{
	"data_source": "vehicleRealtimeFramegrabberData",
	"data_array": [{
		"data_name": "camera_name",
		"data_value": "SciCam"
	}, {
		"data_name": "filename",
		"data_value": "/J2-1229/SciCam/SciCam.20191001_162423276.jpg"
	}, {
		"data_name": "camera_name",
		"data_value": "BrowCam"
	}, {
		"data_name": "filename",
		"data_value": "/J2-1229/BrowCam/BrowCam.20191001_162423276.jpg"
	}, {
		"data_name": "camera_name",
		"data_value": "PilotCam"
	}, {
		"data_name": "filename",
		"data_value": "/J2-1229/PilotCam/PilotCam.20191001_162423276.jpg"
	}, {
		"data_name": "camera_name",
		"data_value": "AftCam"
	}, {
		"data_name": "filename",
		"data_value": "/J2-1229/AftCam/AftCam.20191001_162423276.jpg"
	}],
	"event_id": "5d937db7fef616043cc12b8c",
	"id": "5d937db9fef616043cc12b8f"
}`

Not sure if this ever worked, but currently I'm working on minimally tweaking the UI to get these stills visible from those events.

**Comment by rgov** [comment]
Yep exactly. It never worked previously.

Instead of tweaking the UI can we just translate the `SulisCam` database entries to `vehicleRealtimeFramegrabberData` entries?

**Comment by mcschwartzman** [comment]
Yeah I found that mapper. I'm using the `makeFakeAuxDataForEventOptions` to play around with the data. Maybe later we can do a bulk edit of the existing sulis data

**Comment by mcschwartzman** [comment]
Short-term fix does the trick:

![image](https://user-images.githubusercontent.com/26795373/211651633-16d24e0f-1878-41a1-95ac-0a654197bca2.png)

For a mid-term fix, I'll try writing a little script that creates some aux data entries for the existing sulis events.

For a long-term fix, I think all we need to do is edit the script to create an aux data entry for each sulis still. I _**think**_ the event_export api will handle the rest

**Comment by rgov** [comment]
Nicely done 🙌 Can you open a new issue to track modifying the import script to create aux_data instead of using event options, applying migrations to existing data, and removing the workaround in the rendering code?

**Comment by mcschwartzman** [comment]
https://github.com/WHOIGit/ndsf-sealog-scripts/issues/1
